tutorials/roostats/rs301_splot.py

```diff
@@ -0,0 +1,318 @@
+    # we leave the width of the Z free during the fit in this example.
+
+    # isolation model for Z.  Only used to generate toy MC.
+    # the exponential is of the form exp(c*x).  If we want
+    # the isolation to decay an e-fold every R GeV, we use
+    # c = -1/R.
+    zIsolDecayConst = ROOT.RooConstVar("zIsolDecayConst", "z isolation decay  constant", -1)
+    zIsolationModel = ROOT.RooExponential("zIsolationModel", "z isolation model", isolation, zIsolDecayConst)
+
+    # make the combined Z model
+    zModel = ROOT.RooProdPdf("zModel", "2-d model for Z", ROOT.RooArgSet(mZModel, zIsolationModel))
+
+    # --------------------------------------
+    # make QCD model
+
+    print(f"make qcd model")
+    # mass model for QCD.
+    # the exponential is of the form exp(c*x).  If we want
+    # the mass to decay an e-fold every R GeV, we use
+    # c = -1/R.
+    # We can leave this parameter free during the fit.
+    qcdMassDecayConst = ROOT.RooRealVar(
+        "qcdMassDecayConst", "Decay const for QCD mass spectrum", -0.01, -100, 100, "1/GeV"
+    )
+    qcdMassModel = ROOT.RooExponential("qcdMassModel", "qcd Mass Model", invMass, qcdMassDecayConst)
+
+    # isolation model for QCD.  Only used to generate toy MC
+    # the exponential is of the form exp(c*x).  If we want
+    # the isolation to decay an e-fold every R GeV, we use
+    # c = -1/R.
+    qcdIsolDecayConst = ROOT.RooConstVar("qcdIsolDecayConst", "Et resolution constant", -0.1)
+    qcdIsolationModel = ROOT.RooExponential("qcdIsolationModel", "QCD isolation model", isolation, qcdIsolDecayConst)
+
+    # make the 2-d model
+    qcdModel = ROOT.RooProdPdf("qcdModel", "2-d model for QCD", ROOT.RooArgSet(qcdMassModel, qcdIsolationModel))
+
+    # --------------------------------------
+    gws = object
+    # combined model
+
+    # These variables represent the number of Z or QCD events
+    # They will be fitted.
+    zYield = ROOT.RooRealVar("zYield", "fitted yield for Z", 500, 0.0, 5000)
+    qcdYield = ROOT.RooRealVar("qcdYield", "fitted yield for QCD", 1000, 0.0, 10000)
+
+    # now make the combined models
+    print(f"make full model")
+    model = ROOT.RooAddPdf("model", "z+qcd background models", (zModel, qcdModel), (zYield, qcdYield))
+    massModel = ROOT.RooAddPdf("massModel", "z+qcd invariant mass model", (mZModel, qcdMassModel), (zYield, qcdYield))
+
+    # interesting for debugging and visualizing the model
+    model.graphVizTree("fullModel.dot")
+
+    print(f"import model: ")
+    model.Print()
+
+    object.Import(model)
+
+    try:
+        object.Import(massModel, RecycleConflictNodes=True)
+        print("massModel was imported to workspace resolving ConflictNodes...")
+    except AttributeError:
+        print("Model name conflict! Choose an appropiate action!.")
+
+
+# ____________________________________
+def AddData(object):
+
+    # Add a toy dataset
+
+    # get what we need out of the workspace to make toy data
+    model = object.pdf("model")
```

Comment:
```suggestion
    model = object["model"]
```
We have a Pythonization for the RooWorkspace that everytime you're retrieving something from the workspace (obj, pdf, data, function, or var), you can just use square brackets. Can this be done in all of the file?